src/screens/VisualTests/VisualTests.tsx

```diff
@@ -261,26 +288,48 @@ export const VisualTests = ({
     [reviewTest]
   );
 
-  const build = getFragment(FragmentBuildFields, data?.project?.lastBuild);
-  const isOutdated = build && build.uncommittedHash !== gitInfo.uncommittedHash;
+  const nextBuild = getFragment(FragmentNextBuildFields, data?.project?.lastBuild);
+  // Before we set the storyInfo, we use the nextBuild for story data
+  const storyBuild = getFragment(
+    FragmentStoryBuildFields,
+    data?.storyBuild ?? data?.project?.lastBuild
+  );
 
+  // We always set status to the next build's status, as when we change to a new story we'll see
+  // the next builds
   const buildStatusUpdate =
-    build &&
-    "testsForStatus" in build &&
-    testsToStatusUpdate(getFragment(FragmentStatusTestFields, build.testsForStatus.nodes));
+    nextBuild &&
+    "testsForStatus" in nextBuild &&
+    testsToStatusUpdate(getFragment(FragmentStatusTestFields, nextBuild.testsForStatus.nodes));
 
   useEffect(() => {
     if (buildStatusUpdate) updateBuildStatus(buildStatusUpdate);
     // We use the stringified version of buildStatusUpdate to do a deep diff
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [JSON.stringify(buildStatusUpdate), updateBuildStatus]);
 
-  const [settingsVisible, setSettingsVisible] = useState(false);
-  const [warningsVisible, setWarningsVisible] = useState(false);
-  const [baselineImageVisible, setBaselineImageVisible] = useState(false);
-  const toggleBaselineImage = () => setBaselineImageVisible(!baselineImageVisible);
+  // Ensure we are holding the right story build
+  const hasNextBuild = !!nextBuild;
+  useEffect(() => {
+    if (!hasNextBuild) return;
 
-  if (!build || error) {
+    // We always want storyBuildInfo set if we can
+    if (!storyBuildInfo) {
+      setStoryBuildInfo({ storyId, buildId: nextBuild.id });
+
+      // If you change story, we reset storyBuildInfo to the next build (you see the next build)
+    } else if (storyBuildInfo.storyId !== storyId) {
+      setStoryBuildInfo({ storyId, buildId: nextBuild.id });
+    }
```

Comment:
This seems inefficient, `setStoryBuildInfo` updates `storyBuildInfo` which is a dependency of the useEffect. I think we need to use the setState callback variant here. It also seems both cases set the same value?

```js
setStoryBuildInfo((storyBuildInfo) => {
  return (!storyBuildInfo || storyBuildInfo.storyId !== storyId)
    ? { storyId, buildId: nextBuild.id }
    : storyBuildInfo;
})
```

This way you can drop `storyBuildInfo` from the deps array.